ssl/ssl_ciph.c

```diff
@@ -1495,147 +1495,149 @@ STACK_OF(SSL_CIPHER) *ssl_create_cipher_list(SSL_CTX *ctx,
-     * of ciphers themselves 2) aliases for groups of ciphers.
-     * For 1) we need the available ciphers and for 2) the cipher
-     * groups of cipher_aliases added together in one list (otherwise
-     * we would be happy with just the cipher_aliases table).
-     */
-    num_of_group_aliases = OSSL_NELEM(cipher_aliases);
-    num_of_alias_max = num_of_ciphers + num_of_group_aliases + 1;
-    ca_list = OPENSSL_malloc(sizeof(*ca_list) * num_of_alias_max);
-    if (ca_list == NULL) {
-        OPENSSL_free(co_list);
-        return NULL;          /* Failure */
-    }
-    ssl_cipher_collect_aliases(ca_list, num_of_group_aliases,
-                               disabled_mkey, disabled_auth, disabled_enc,
-                               disabled_mac, head);
+        /*
+        * We also need cipher aliases for selecting based on the rule_str.
+        * There might be two types of entries in the rule_str: 1) names
+        * of ciphers themselves 2) aliases for groups of ciphers.
+        * For 1) we need the available ciphers and for 2) the cipher
+        * groups of cipher_aliases added together in one list (otherwise
+        * we would be happy with just the cipher_aliases table).
+        */
+        num_of_group_aliases = OSSL_NELEM(cipher_aliases);
+        num_of_alias_max = num_of_ciphers + num_of_group_aliases + 1;
+        ca_list = OPENSSL_malloc(sizeof(*ca_list) * num_of_alias_max);
+        if (ca_list == NULL) {
+            OPENSSL_free(co_list);
+            return NULL;          /* Failure */
+        }
+        ssl_cipher_collect_aliases(ca_list, num_of_group_aliases,
+                                disabled_mkey, disabled_auth, disabled_enc,
+                                disabled_mac, head);
 
-    /*
-     * If the rule_string begins with DEFAULT, apply the default rule
-     * before using the (possibly available) additional rules.
-     */
-    ok = 1;
-    rule_p = rule_str;
-    if (HAS_PREFIX(rule_str, "DEFAULT")) {
-        ok = ssl_cipher_process_rulestr(OSSL_default_cipher_list(),
-                                        &head, &tail, ca_list, c);
-        rule_p += 7;
-        if (*rule_p == ':')
-            rule_p++;
-    }
+        /*
+        * If the rule_string begins with DEFAULT, apply the default rule
+        * before using the (possibly available) additional rules.
+        */
+        ok = 1;
+        rule_p = rule_str;
+        if (HAS_PREFIX(rule_str, "DEFAULT")) {
+            ok = ssl_cipher_process_rulestr(OSSL_default_cipher_list(),
+                                            &head, &tail, ca_list, c);
+            rule_p += 7;
+            if (*rule_p == ':')
+                rule_p++;
+        }
 
-    if (ok && (rule_p[0] != '\0'))
-        ok = ssl_cipher_process_rulestr(rule_p, &head, &tail, ca_list, c);
+        if (ok && (rule_p[0] != '\0'))
+            ok = ssl_cipher_process_rulestr(rule_p, &head, &tail, ca_list, c);
```

Comment:
I believe this means the `@SECLEVEL=x` in the cipher string won't be processed for QUIC.

Reply:
Fixed